// @gamestdio/keycode/index.js
"use strict";
exports.BACKSPACE = 8;
exports.TAB = 9;
exports.ENTER = 13;
exports.SHIFT = 16;
exports.CTRL = 17;
exports.ALT = 18;
exports.PAUSE = 19;
exports.CAPSLOCK = 20;
exports.ESCAPE = 27;
exports.SPACE = 32;
exports.PAGEUP = 33;
exports.PAGEDOWN = 34;
exports.END = 35;
exports.HOME = 36;
exports.LEFT = 37;
exports.UP = 38;
exports.RIGHT = 39;
exports.DOWN = 40;
exports.INSERT = 45;
exports.DELETE = 46;
// numbers
exports.KEY_0 = 48;
exports.KEY_1 = 49;
exports.KEY_2 = 50;
exports.KEY_3 = 51;
exports.KEY_4 = 52;
exports.KEY_5 = 53;
exports.KEY_6 = 54;
exports.KEY_7 = 55;
exports.KEY_8 = 56;
exports.KEY_9 = 57;
// alphabet
exports.A = 65;
exports.B = 66;
exports.C = 67;
exports.D = 68;
exports.E = 69;
exports.F = 70;
exports.G = 71;
exports.H = 72;
exports.I = 73;
exports.J = 74;
exports.K = 75;
exports.L = 76;
exports.M = 77;
exports.N = 78;
exports.O = 79;
exports.P = 80;
exports.Q = 81;
exports.R = 82;
exports.S = 83;
exports.T = 84;
exports.U = 85;
exports.V = 86;
exports.W = 87;
exports.X = 88;
exports.Y = 89;
exports.Z = 90;
exports.SELECT = 93;
exports.NUMPAD_0 = 96;
exports.NUMPAD_1 = 97;
exports.NUMPAD_2 = 98;
exports.NUMPAD_3 = 99;
exports.NUMPAD_4 = 100;
exports.NUMPAD_5 = 101;
exports.NUMPAD_6 = 102;
exports.NUMPAD_7 = 103;
exports.NUMPAD_8 = 104;
exports.NUMPAD_9 = 105;
exports.MULTIPLY = 106;
exports.ADD = 107;
exports.SUBTRACT = 109;
exports.DECIMALPOINT = 110;
exports.DIVIDE = 111;
// F1~F2
exports.F1 = 112;
exports.F2 = 113;
exports.F3 = 114;
exports.F4 = 115;
exports.F5 = 116;
exports.F6 = 117;
exports.F7 = 118;
exports.F8 = 119;
exports.F9 = 120;
exports.F10 = 121;
exports.F11 = 122;
exports.F12 = 123;
// etc / accents
exports.NUMLOCK = 144;
exports.SCROLLLOCK = 145;
exports.SEMICOLON = 186;
exports.EQUALSIGN = 187;
exports.COMMA = 188;
exports.DASH = 189;
exports.PERIOD = 190;
exports.FORWARDSLASH = 191;
exports.GRAVEACCENT = 192;
exports.OPENBRACKET = 219;
exports.BACKSLASH = 220;
exports.CLOSEBRAKET = 221;
exports.SINGLEQUOTE = 222;
