// colyseus/lib/serializer/FossilDeltaSerializer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const fossilDelta = require("fossil-delta");
const msgpack = require("notepack.io");
const jsonPatch = require("fast-json-patch"); // this is only used for debugging patches
const Debug_1 = require("../Debug");
class FossilDeltaSerializer {
    constructor() {
        this.id = "fossil-delta";
    }
    reset(newState) {
        this.previousState = newState;
        this.previousStateEncoded = msgpack.encode(this.previousState);
    }
    getData() {
        return this.previousStateEncoded;
    }
    hasChanged(newState) {
        const currentState = newState;
        const currentStateEncoded = msgpack.encode(currentState);
        const changed = !currentStateEncoded.equals(this.previousStateEncoded);
        if (changed) {
            this.patches = fossilDelta.create(this.previousStateEncoded, currentStateEncoded);
            //
            // debugging
            //
            if (Debug_1.debugPatch.enabled) {
                Debug_1.debugPatch('%d bytes, %j', this.patches.length, jsonPatch.compare(msgpack.decode(this.previousStateEncoded), currentState));
            }
            this.previousState = currentState;
            this.previousStateEncoded = currentStateEncoded;
        }
        return changed;
    }
    getPatches() {
        return this.patches;
    }
}
exports.FossilDeltaSerializer = FossilDeltaSerializer;
